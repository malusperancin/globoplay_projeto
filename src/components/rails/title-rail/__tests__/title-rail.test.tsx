/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-node-access */
import React from "react";
import { render } from "@testing-library/react";
import TitleRail from "../title-rail";
import TitleRailType from "../title-rail-type";
import TitleCardType from "../../../cards/title-card/title-card-type";

describe("TitleRail", () => {
  const cards: TitleCardType[] = Array(10).map((_, index) => ({
    titleId: "teste",
    headline: "teste",
    description: "teste",
    poster: {
      tv: "teste",
    },
    backgroundUrl: "teste",
    selfRatedContent: true,
    contentRating: "teste",
    type: "teste",
    technicalSpecs: {
      resolutions: [],
    },
    genresNames: [],
    format: "teste",
    releaseYear: 2022,
    contentBrand: {
      trimmedLogo: "teste",
      name: "teste",
    },
    originalContent: true,
    subset: {
      slug: "teste",
    },
  }));

  const list: TitleRailType = {
    title: "teste",
    contentType: "TITLE",
    index: 0,
    cards,
  };

  const titleComponent = (
    <TitleRail
      data={list}
      isFocused={true}
      onCardFocus={jest.fn()}
      hide={false}
    />
  );

  const titleHideComponent = (
    <TitleRail
      data={list}
      isFocused={true}
      onCardFocus={jest.fn()}
      hide={true}
    />
  );

  it("should render without throwing an error", () => {
    const { container } = render(titleComponent);

    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });

  it("should hide component", () => {
    const { container } = render(titleHideComponent);
    expect(container).toBeEmptyDOMElement();
  });

  it("should match snapshot", () => {
    const { container } = render(titleHideComponent);
    expect(container).toBeEmptyDOMElement();
  });
});
