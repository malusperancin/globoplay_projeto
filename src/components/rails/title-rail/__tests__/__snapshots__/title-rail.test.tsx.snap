// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleRail should render without throwing an error 1`] = `
<div>
  <div>
    <p
      class="title"
    >
       
      teste
       
    </p>
    <div
      class="rail"
    />
  </div>
</div>
`;
